pkg/config/config.go

```diff
@@ -163,85 +169,87 @@ func init() {
+	BindEnvAndSetDefault("forwarder_timeout", 20)
+	BindEnvAndSetDefault("forwarder_retry_queue_max_size", 30)
 	BindEnvAndSetDefault("forwarder_num_workers", 1)
 	// Dogstatsd
-	Datadog.SetDefault("use_dogstatsd", true)
-	Datadog.SetDefault("dogstatsd_port", 8125)          // Notice: 0 means UDP port closed
-	Datadog.SetDefault("dogstatsd_buffer_size", 1024*8) // 8KB buffer
-	Datadog.SetDefault("dogstatsd_non_local_traffic", false)
-	Datadog.SetDefault("dogstatsd_socket", "") // Notice: empty means feature disabled
-	Datadog.SetDefault("dogstatsd_stats_port", 5000)
-	Datadog.SetDefault("dogstatsd_stats_enable", false)
-	Datadog.SetDefault("dogstatsd_stats_buffer", 10)
-	Datadog.SetDefault("dogstatsd_expiry_seconds", 300)
-	Datadog.SetDefault("dogstatsd_origin_detection", false) // Only supported for socket traffic
-	Datadog.SetDefault("dogstatsd_so_rcvbuf", 0)
-	Datadog.SetDefault("statsd_forward_host", "")
-	Datadog.SetDefault("statsd_forward_port", 0)
+	BindEnvAndSetDefault("use_dogstatsd", true)
+	BindEnvAndSetDefault("dogstatsd_port", 8125)          // Notice: 0 means UDP port closed
+	BindEnvAndSetDefault("dogstatsd_buffer_size", 1024*8) // 8KB buffer
+	BindEnvAndSetDefault("dogstatsd_non_local_traffic", false)
+	BindEnvAndSetDefault("dogstatsd_socket", "") // Notice: empty means feature disabled
+	BindEnvAndSetDefault("dogstatsd_stats_port", 5000)
+	BindEnvAndSetDefault("dogstatsd_stats_enable", false)
+	BindEnvAndSetDefault("dogstatsd_stats_buffer", 10)
+	BindEnvAndSetDefault("dogstatsd_expiry_seconds", 300)
+	BindEnvAndSetDefault("dogstatsd_origin_detection", false) // Only supported for socket traffic
+	BindEnvAndSetDefault("dogstatsd_so_rcvbuf", 0)
+	BindEnvAndSetDefault("statsd_forward_host", "")
+	BindEnvAndSetDefault("statsd_forward_port", 0)
 	BindEnvAndSetDefault("statsd_metric_namespace", "")
 	// Autoconfig
-	Datadog.SetDefault("autoconf_template_dir", "/datadog/check_configs")
-	Datadog.SetDefault("exclude_pause_container", true)
-	Datadog.SetDefault("ac_include", []string{})
-	Datadog.SetDefault("ac_exclude", []string{})
+	BindEnvAndSetDefault("autoconf_template_dir", "/datadog/check_configs")
+	BindEnvAndSetDefault("exclude_pause_container", true)
+	BindEnvAndSetDefault("ac_include", []string{})
+	BindEnvAndSetDefault("ac_exclude", []string{})
 
 	// Docker
 	BindEnvAndSetDefault("docker_query_timeout", int64(5))
-	Datadog.SetDefault("docker_labels_as_tags", map[string]string{})
-	Datadog.SetDefault("docker_env_as_tags", map[string]string{})
-	Datadog.SetDefault("kubernetes_pod_labels_as_tags", map[string]string{})
-	Datadog.SetDefault("kubernetes_pod_annotations_as_tags", map[string]string{})
-	Datadog.SetDefault("kubernetes_node_labels_as_tags", map[string]string{})
+	BindEnvAndSetDefault("docker_labels_as_tags", map[string]string{})
+	BindEnvAndSetDefault("docker_env_as_tags", map[string]string{})
+	BindEnvAndSetDefault("kubernetes_pod_labels_as_tags", map[string]string{})
```

Comment:
For `GetStringMapString` to work correctly these env vars need to be `json`. Should that be documented anywhere?

Reply:
Oh yes, I'll add that

Reply:
done :)